app/src/test/java/org/simple/clinic/home/overdue/OverdueLogicTest.kt

```diff
@@ -115,14 +110,10 @@ class OverdueLogicTest {
   }
 
   private fun setupController() {
-    whenever(userSession.loggedInUser()).thenReturn(Observable.just(user.toOptional()))
-    whenever(facilityRepository.currentFacility(user)).thenReturn(Observable.just(facility))
-
     val effectHandler = OverdueEffectHandler(
         schedulers = TestSchedulersProvider.trampoline(),
-        userSession = userSession,
-        facilityRepository = facilityRepository,
         appointmentRepository = repository,
+        currentFacility = Lazy { facility },
```

Comment:
It's when we need to defer invoking the method until it's required.

Here, we know that the current facility will never change while in this screen and it is pointless to query it from the database each time it is required. However, we cannot also query it at the time of dependency injection because that happens at the UI thread during view inflation and we cannot make database queries there.

This is why we will instead use `dagger.Lazy` to represent these dependencies so the actual database query will only happen once in that screen.